common4j/src/main/com/microsoft/identity/common/java/cache/SharedPreferencesAccountCredentialCache.java

```diff
@@ -100,113 +107,150 @@ public SharedPreferencesAccountCredentialCache(
         Logger.verbose(TAG, "Init: " + TAG);
         mSharedPreferencesFileManager = sharedPreferencesFileManager;
         mCacheValueDelegate = accountCacheValueDelegate;
+        new Thread(() -> load()).start();
```

Comment:
We should probably use an ExecutorService to manage these threads

Reply:
This is a single one-shot job at the load of the cache.  I modeled this largely after how Android source's SharedPreferences initially loads the XML file into cached memory.

https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/app/SharedPreferencesImpl.java#134

Is there an existing Executor in use as a general thread pool that I could post this job to in AndroidCommon land without spinning up a new thread?  Priority #1 was not keeping a new thread worth of resources around for a long time to do this one job, but not spinning one up in the first place would be an improvement.

Reply:
We have some in [CommandDisptacher ](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java#L91)that we use...these are private and specific to that class though.

You may want to just create an executor inside `SharedPreferencesAccountCredentialCache.java` and delegate your load operations to that executor.

Reply:
I looked into this, and the thread from newSingleThreadExecutor will stick around forever, which we don't want for this single job.  For running a single job, without having a better long lived queue to share at a wider level, I think this is the best we can do.